### DISCUSSION TITLE
Prevent page reload when User navigates back to the page using the Browser Back Button

### DISCUSSION THREAD

**Discussion by tippfelher** [vercel/next.js#46701] [Help]
### Summary

I have a very simple navigation flow:

1) User visits "`/`" (`index.tsx` in my repro)

2) "`/`" loads search results (or whatever) from a remote service and displays them

3) User clicks on one of the search results which leads to the search result `/post` (`post.tsx` in my repro)

Problem:

4) When the user uses the Browser back button on `/post` the search results are yet again fetched from the remote service.

Question:
How can I prevent the reload in 4)? I just want to represent the page in the way the user left it -- ideally including the users scrollbar position etc.

### Example

https://github.com/tippfelher/dp-nextjs-backbutton

**Comment by ciruz** [comment]
I would take a look at `react-query` or `SWR`, they have great caching options, with them it's pretty simple (out of the box) to achieve the behavior you want.
